src/test/java/uk/gov/pay/products/service/PaymentCreatorTest.java

```diff
@@ -0,0 +1,191 @@
+package uk.gov.pay.products.service;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.pay.products.client.publicapi.PaymentRequest;
+import uk.gov.pay.products.client.publicapi.PaymentResponse;
+import uk.gov.pay.products.client.publicapi.PublicApiResponseErrorException;
+import uk.gov.pay.products.client.publicapi.PublicApiRestClient;
+import uk.gov.pay.products.client.publicapi.model.Link;
+import uk.gov.pay.products.client.publicapi.model.Links;
+import uk.gov.pay.products.matchers.PaymentEntityMatcher;
+import uk.gov.pay.products.matchers.PaymentRequestMatcher;
+import uk.gov.pay.products.model.Payment;
+import uk.gov.pay.products.persistence.dao.PaymentDao;
+import uk.gov.pay.products.persistence.dao.ProductDao;
+import uk.gov.pay.products.persistence.entity.PaymentEntity;
+import uk.gov.pay.products.persistence.entity.ProductEntity;
+import uk.gov.pay.products.service.transaction.TransactionFlow;
+import uk.gov.pay.products.util.PaymentStatus;
+
+import java.util.Optional;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.*;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.hamcrest.MockitoHamcrest.argThat;
+import static uk.gov.pay.products.util.PaymentStatus.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PaymentCreatorTest {
+
+    @Mock
+    private ProductDao productDao;
+
+    @Mock
+    private PaymentDao paymentDao;
+
+    @Mock
+    private PublicApiRestClient publicApiRestClient;
+
+    private PaymentCreator paymentCreator;
+
+    @Before
+    public void setup() throws Exception {
+        LinksDecorator linksDecorator = new LinksDecorator("https://products.url", "https://products-ui.url");
+        paymentCreator = new PaymentCreator(TransactionFlow::new, productDao, paymentDao, publicApiRestClient, linksDecorator);
+    }
+
+    @Test
+    public void shouldCreateASuccessfulPayment() throws Exception {
+        int productId = 1;
+        String productExternalId = "product-external-id";
+        long productPrice = 100l;
```

Comment:
Lower case `l` in `long` should be `L` to avoid confusion

Reply:
👍